keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1525 @@
+
+Introducing pod-level resource requests and limits offers a more simplified
+resource management for multi-container pods as it is easier to gauge theß
+collective resource usage of all containers in a pod rather than predicting each
+container’s individual needs. This proposal seeks to support pod-level resource
+management, enabling Kubernetes to control the total resource consumption of the
+pod, relieving users from the burden of meticulously configuring resources for
+each container. This simplified approach is particularly valuable for tightly
+coupled applications, and it will enable Kubernetes to leverage the underlying
+cgroup resource management mechanisms more effectively for better resource
+utilization for bursty workloads. 
+
+This new feature of resource specification at pod-level would complement, not
+replace, the existing container-level limits, providing users with greater
+flexibility.
+
+### Goals
+
+This proposal aims to:
+
+1. Extend Pod API to allow specifying resource spec (requests and limits) at the
+   pod level. 
+2. Make this feature compatible with existing usage of container level
+   requests/limits, and other features like topology manager, memory manager,
+   etc.
+
+### Non Goals
+
+1. This KEP focuses on the core resource types of CPU and memory. It doesn’t
+intend to address other resource types (e.g. GPU, storage, network bandwidth) at
+the pod level in this phase. However, it could be considered in future extensions of
+the KEP.
+
+## Proposal
+
+
+### User Stories
+
+
+#### Simplified Resource Management
+
+There are cases when it is challenging to decide a-priori how much resources to
+allocate to each container in a pod. For instance, consider a development
+environment implemented as a Kubernetes Pod with multiple sidecar containers for
+tools, and a regular container for web-based IDE. The IDE container may have a
+fixed memory limit, but the resource needs of the tool containers can fluctuate
+depending on the developer's activities.
+
+The proposed Pod Level Resource Spec feature addresses this by allowing all
+containers within the pod to share a defined resource pool. In the example
+provided, even though the IDE has its own resource constraints, tool1 and tool 2
+containers can dynamically utilize any remaining resources within the pod's
+overall limit. 
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    run: ide
+  name: myide
+spec:
+  resources:
+    limits:
+      memory: "1024Mi"
+      cpu: "4"
+  initContainers:
+    - image: shell
+      name: debian:buster
+    - image: tool1
+      name: first-tool:latest
+    - image: tool2
+      name: second-tool:latest
+  containers:
+    - name: ide
+      image: theia:latest
+      resources:
+        requests:
+          memory: 128M
+          cpu: "0.5"
+        limits:
+          memory: 256M
+          cpu: "1"
+```
+
+Without this feature, the developers will need to estimate beforehand how much
```

Comment:
the pod spec above shows 3 init containers that will be completed by the time `ide` will start. Was it meant to be sidecars?